### PR TITLE
Back command improvement

### PR DESCRIPTION
**Pull request by Akiranya**
Implemented #403 

It's my first attempt to contribute to this project so the changes may not be optimal.

I'm happy to see if there's anything can be improved!

### CODE REVIEW

File: common/src/main/java/net/william278/huskhomes/command/BackCommand.java [hunk 1]
```diff
@@ -36,6 +36,7 @@ public class BackCommand extends InGameCommand {
     protected BackCommand(@NotNull HuskHomes plugin) {
         super("back", List.of(), "", plugin);
         addAdditionalPermissions(Map.of("death", false));
+        addAdditionalPermissions(Map.of("previous", false));
```
Comment (WiIIiam278) [inline common/src/main/java/net/william278/huskhomes/command/BackCommand.java:39]:
```suggestion
        addAdditionalPermissions(Map.of(
            "death", false,
            "previous", false
        ));
```
`addAdditionalPermissions` accepts a map, so we can simply add more pairs of values to the first `Map#of` call to generate a map with two key-value pairs :)

**Review by WiIIiam278 — CHANGES_REQUESTED** [review]
Thank you for your work -- this is a great first PR! There's only one minor tweak I would make to this related to code style. Other than that, perfect!

Side note on the dependencies: Don't worry, it's not just you, Jitpack is extremely tempremental!

**Comment by Akiranya** [comment]
If I don't get it wrong, the piece of code:
```
addAdditionalPermissions(Map.of(
            "death", false,
            "previous", false
        ));
```

is saying that:

Player don't have `death` and `previous` permissions by default because the map values are `false`.

But in game the fact is that players have these permissions by default. Permission check output of LuckPerms:
```
> deop Nailm
[11:49:46 INFO]: Nothing changed. The player is not an operator
> lp user Nailm permission check huskhomes.command.back.death
[11:49:54 INFO]: [LP] Permission information for huskhomes.command.back.death:
[11:49:54 INFO]: [LP] - nailm does not have huskhomes.command.back.death set.
[11:49:54 INFO]: [LP] - nailm does not inherit huskhomes.command.back.death.
[11:49:54 INFO]: [LP] 
[11:49:54 INFO]: [LP] Permission check for huskhomes.command.back.death:
[11:49:54 INFO]: [LP]     Result: true
[11:49:54 INFO]: [LP]     Processor: bukkit.DefaultPermissionMapProcessor
[11:49:54 INFO]: [LP]     Cause: None
[11:49:54 INFO]: [LP]     Context: (dimension-type=overworld) (essentials:afk=true) (essentials:jailed=false) (essentials:muted=false) (essentials:vanished=false) (gamemode=survival) (server=hub) (server-group=hub) (server-id=hub) (server-name=斯卡尔德港湾) (towny:insideownplot=false) (towny:insideowntown=true) (towny:insidetown=true) (towny:king=false) (towny:mayor=true) (towny:resident=true) (towny:town=roucheng) (world=world)
> lp user Nailm permission check huskhomes.command.back.previous
[11:49:57 INFO]: [LP] Permission information for huskhomes.command.back.previous:
[11:49:57 INFO]: [LP] - nailm does not have huskhomes.command.back.previous set.
[11:49:57 INFO]: [LP] - nailm does not inherit huskhomes.command.back.previous.
[11:49:57 INFO]: [LP] 
[11:49:57 INFO]: [LP] Permission check for huskhomes.command.back.previous:
[11:49:57 INFO]: [LP]     Result: true
[11:49:57 INFO]: [LP]     Processor: bukkit.DefaultPermissionMapProcessor
[11:49:57 INFO]: [LP]     Cause: None
[11:49:57 INFO]: [LP]     Context: (dimension-type=overworld) (essentials:afk=true) (essentials:jailed=false) (essentials:muted=false) (essentials:vanished=false) (gamemode=survival) (server=hub) (server-group=hub) (server-id=hub) (server-name=斯卡尔德港湾) (towny:insideownplot=false) (towny:insideowntown=true) (towny:insidetown=true) (towny:king=false) (towny:mayor=true) (towny:resident=true) (towny:town=roucheng) (world=world)
```

The `Result: true` presents in both of the checks.

File: common/src/main/java/net/william278/huskhomes/command/BackCommand.java (hunk 1 above)
Comment (Akiranya) [reply common/src/main/java/net/william278/huskhomes/command/BackCommand.java]: Nice catch! Done

But I have another concern about the permission resolution (I've commeted it on this issue)

**Comment by Akiranya** [comment]
This is a screenshot of the checks (which is easy to read):

image

**Comment by Akiranya** [comment]
lol nevermind, I just found it worked exactly the reverse way as I thought

**Comment by WiIIiam278** [comment]
> lol nevermind, I just found it worked exactly the reverse way as I thought

Yeah the system here is kind of confusing; in effect, the boolean here is controlling whether it is an `operator-restricted` command; so in this case both should be false by default as the two should not be operator restricted.

**Review by WiIIiam278 — APPROVED** [review]

**Pull request merged**